### PR TITLE
[15.0][MIG] account_reporting_weight: Migration to 15.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 15.0

Please @pedrobaeza and @cesar-tecnativa can you review it?

@Tecnativa TT36435

### CODE REVIEW

File: account_reporting_weight/reports/account_invoice_report.py [hunk 1]
```diff
@@ -13,12 +13,13 @@ class AccountInvoiceReport(models.Model):
...
         select_str = super()._select()
         select_str += """
-            , SUM(product.weight * (
-                CASE
-                WHEN move.type IN ('in_invoice','out_refund','in_receipt') THEN -1
-                ELSE 1 END
-            ) * line.quantity
-            / uom_line.factor * uom_template.factor)
-            as weight
+            , COALESCE(
```
Comment (pedrobaeza) [inline account_reporting_weight/reports/account_invoice_report.py:16]: Why there's not `SUM` now and why `COALESCE` is needed?

Comment (victoralmau) [reply account_reporting_weight/reports/account_invoice_report.py:16]: `SUM` required adding `GROUP BY`.
I have added `COALESCE` to not show NULL if, for example, a line has no product.

I have functionally tested with different use cases and the weight value is correct.

Is there any other way to make the change?

**Review by pedrobaeza — APPROVED** [review]
It's OK this way. Odoo now process the reports other way.

**Comment by victoralmau** [comment]
Ping @cesar-tecnativa

File: account_reporting_weight/__manifest__.py [hunk 2]
```diff
@@ -1,9 +1,9 @@
-# Copyright 2017 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2017 Tecnativa - Pedro M. Baeza
```
Comment (cesar-tecnativa) [inline account_reporting_weight/__manifest__.py:1]: Copyright 2012

File: account_reporting_weight/readme/CONTRIBUTORS.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+* `Tecnativa <https://www.tecnativa.com>`_:
+
+  * Pedro M. Baeza
+  * David Vidal
+  * Sergio Teruel
+  * Carlos Roca
```
Comment (cesar-tecnativa) [inline account_reporting_weight/readme/CONTRIBUTORS.rst:6]: Add @victoralmau

File: account_reporting_weight/readme/USAGE.rst [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#. Go to *Invoicing > Reporting > Management > Invoices*.
```
Comment (cesar-tecnativa) [inline account_reporting_weight/readme/USAGE.rst]: #. Go to *Invoicing > Reporting > Management > Invoice Analysis*.

File: account_reporting_weight/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+{
+    "name": "Weights in the invoices analysis view",
+    "version": "15.0.1.0.0",
+    "author": "Tecnativa," "Odoo Community Association (OCA)",
+    "category": "Inventory, Logistics, Warehousing",
+    "development_status": "Production/Stable",
+    "license": "AGPL-3",
+    "website": "https://github.com/OCA/account-invoice-reporting",
+    "depends": ["sale"],
```
Comment (cesar-tecnativa) [inline account_reporting_weight/__manifest__.py]: "depends": ["account"],

**Review by cesar-tecnativa — CHANGES_REQUESTED** [review]

File: account_reporting_weight/__manifest__.py (hunk 2 above)
Comment (victoralmau) [reply account_reporting_weight/__manifest__.py:1]: I do not understand what you are suggesting, only the email is deleted to avoid spam.

Comment (cesar-tecnativa) [reply account_reporting_weight/__manifest__.py:1]: The year... but it is ok

**Comment by pedrobaeza** [comment]
@victoralmau you haven't said that the changes are done

**Review by cesar-tecnativa — APPROVED** [review]

**Pull request merged**